pkgs/development/ocaml-modules/re/default.nix

```diff
@@ -2,15 +2,13 @@
 
 buildDunePackage rec {
   pname = "re";
-  version = "1.9.0";
+  version = "1.10.4";
 
-  minimumOCamlVersion = "4.02";
-
-  useDune2 = lib.versionAtLeast ocaml.version "4.08";
```

Comment:
Can you motivate this change? As is, it’s just breaking things…

Reply:
Changed to `minimalOCamlVersion` which I think is the new/correct attr(?) and I also tried building on these version:
```
➜  nixpkgs git:(ulrikstrid--ocaml-re-1_10_4) ✗ nix build .#ocaml-ng.ocamlPackages_4_03.re
➜  nixpkgs git:(ulrikstrid--ocaml-re-1_10_4) ✗ nix build .#ocaml-ng.ocamlPackages_4_10.re
➜  nixpkgs git:(ulrikstrid--ocaml-re-1_10_4) ✗ nix build .#ocaml-ng.ocamlPackages_4_14.re
```
those 3 versions built fine so it doesn't break anything on my machine.

Reply:
Both attributes work together: if you want to use dune 2 for all versions, then raise `minimalOCamlVersion` to 4.08 at least. Otherwise, continue using dune 1 for OCaml < 4.08 as is currently the case.

Reply:
Okay, I'm surprised it built for me of this is wrong. But I can revert the `useDune2` change

Reply:
The problem is not about building the library (this dune 2 can do) but using it with dune 1.

Reply:
Reverted the `useDune2` change, thank you for explaining!